src/actor/npc/ac_npc_shop_master.c

```diff
@@ -0,0 +1,2677 @@
+    shop_master->player_angle = shop_master->npc_class.actor_class.player_angle_y; 
+}
+
+static int aNSC_check_buy_item_single(NPC_SHOP_MASTER_ACTOR *nook, Submenu *menu) {
+
+    mActor_name_t item = menu->item_p[0].item;
+    Submenu_Item_c *item_p = &menu->item_p[0];
+
+    u32 res = aNSC_CHECK_BUY_NORMAL_SUB;
+    u32 quant = 0;
+
+    if (NowPrivate_GetItemCond(item_p->slot_no) & mPr_ITEM_COND_QUEST) {
+        res = aNSC_CHECK_BUY_REFUSE_QUEST_COND;
+    }
+    else if (ITEM_NAME_GET_TYPE(item) == NAME_TYPE_ITEM1 && ITEM_NAME_GET_CAT(item) == ITEM1_CAT_KABU) {
+        if (item == ITM_KABU_SPOILED) {
+            res = aNSC_CHECK_BUY_TAKE_OFF_HANDS;
+            quant = 1;
+        }
+        else if (Common_Get(time).rtc_time.weekday == lbRTC_SUNDAY) {
+            res = aNSC_CHECK_BUY_NO_SUNDAY_TURNIPS;
+        }
+        else {
+            res = aNSC_check_buy_item_sub(&quant, item);
+        }
+    } 
+    else {
+        if (mSP_ItemNo2ItemPrice(item) / SELL_BUY_RATIO == 0) {
+            res = aNSC_CHECK_BUY_TAKE_OFF_HANDS;
+            quant = 1;
+        }
+        else if (ITEM_IS_PAPER(item)) {
+            res = aNSC_check_buy_paper(&quant, item);
+        } else {
+            res = aNSC_check_buy_item_sub(&quant, item);
+        }
+    }
+    
+    nook->counter = quant;
+    return res;
+}
+
+static int aNSC_check_buy_item_plural(NPC_SHOP_MASTER_ACTOR *nook, Submenu *menu) {
+    Submenu_Item_c *selected = menu->item_p;
+
+    u32 num_selected = menu->selected_item_num;
+    int res = aNSC_CHECK_BUY_NORMAL;
+    nook->counter = num_selected;
+
+    while (num_selected) {
+        if (ITEM_NAME_GET_TYPE(selected->item) == NAME_TYPE_ITEM1) {
+            if (ITEM_NAME_GET_CAT(selected->item) == ITEM1_CAT_KABU) {
+                res = aNSC_CHECK_BUY_TURNIPS;
+                break;
+            }
+        }
+        if (mSP_ItemNo2ItemPrice(selected->item) / SELL_BUY_RATIO != 0) {
+            res = aNSC_CHECK_BUY_TURNIPS;
+            break;
+        }
+
+        selected += 1;
+        num_selected -= 1;
+    }
+
+    return res;
+}
+
+static int aNSC_check_buy_item(NPC_SHOP_MASTER_ACTOR *sm, Submenu *menu) {
+    if (menu->selected_item_num == 1) {
+        return aNSC_check_buy_item_single(sm, menu);
+    } else {
+        return aNSC_check_buy_item_plural(sm, menu);
+    }
+}
+
+static int aNSC_check_item_with_ticket(mActor_name_t item) {
+    int type;
+    int cat;
+    int res;
+
+    type = ITEM_NAME_GET_TYPE(item);
+    res = 0;
+
+    switch (ITEM_NAME_GET_TYPE(item)) {
+    case NAME_TYPE_FTR0:
+    case NAME_TYPE_FTR1:
+        return TRUE;
+    case NAME_TYPE_ITEM1:
+        switch (ITEM_NAME_GET_CAT(item)) {
+        case ITEM1_CAT_CLOTH:
+        case ITEM1_CAT_WALL:
+        case ITEM1_CAT_CARPET:
+            res = TRUE;
+            break;
+        case ITEM1_CAT_TOOL:
+            if (item >= ITM_UMBRELLA_START && item <= ITM_UMBRELLA_END - 1) {
+                res = TRUE;
+            }
```

Comment:
This case is missing a break. Unless a case falls through you should include a break or return for all switch cases.